pkg/workers/workers.go

```diff
@@ -188,12 +188,19 @@ func updateRecords(node *masa.OracleNode, workEvent db.WorkEvent) {
 	exists := db.ReadData(node, workEvent.CID)
 	if exists == nil {
 		_ = db.WriteData(node, workEvent.CID, workEvent.Payload)
+		err := node.PubSubManager.Publish(config.TopicWithVersion(config.BlockTopic), workEvent.Payload)
+		if err != nil {
+			logrus.Errorf("Error publishing block: %v", err)
+		}
+
 	}
 
 	var nodeData pubsub.NodeData
 	nodeDataBytes, err := db.GetCache(context.Background(), workEvent.PeerId)
 	if err != nil {
-		nodeData = *node.NodeTracker.GetNodeData(workEvent.PeerId)
+		if workEvent.PeerId != "" {
+			nodeData = *node.NodeTracker.GetNodeData(workEvent.PeerId)
```

Comment:
This dereference is making assumption that we always return a non-nil pointer. While that might be correct for 99% of the cases, it's better to be defensive here and check it what `GetNodeData` is returning is not nil

Reply:
resolved by B